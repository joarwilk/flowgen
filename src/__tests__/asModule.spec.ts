import { compiler, beautify } from "..";
import "../test-matchers";

it("should wrap in module", () => {
  const ts = `
declare export type Test = 'ok' | 'error'
declare type Test2 = 'ok' | 'error'
type Maybe<T> = {type: 'just', value: T} | {type: 'nothing'}
export type Ref<T> = { current: T }

export const ok: number
`;
  const result = compiler.compileDefinitionString(ts, { quiet: true, asModule: "myModule" });
  expect(beautify(result)).toMatchSnapshot();
  expect(result).toBeValidFlowTypeDeclarations();
});

it("should wrap in module with scoped name", () => {
  const ts = `
declare export type Test = 'ok' | 'error'
declare type Test2 = 'ok' | 'error'
type Maybe<T> = {type: 'just', value: T} | {type: 'nothing'}
export type Ref<T> = { current: T }

export const ok: number
`;
  const result = compiler.compileDefinitionString(ts, { quiet: true, asModule: "@company/project" });
  expect(beautify(result)).toMatchSnapshot();
  expect(result).toBeValidFlowTypeDeclarations();
});
