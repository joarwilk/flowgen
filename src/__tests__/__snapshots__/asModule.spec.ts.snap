// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should wrap in module 1`] = `
"declare module \\"myModule\\" {
  declare type Test = \\"ok\\" | \\"error\\";
  declare type Test2 = \\"ok\\" | \\"error\\";
  declare type Maybe<T> =
    | {
        type: \\"just\\",
        value: T,
        ...
      }
    | {
        type: \\"nothing\\",
        ...
      };
  declare type Ref<T> = {
    current: T,
    ...
  };
  declare var ok: number;
}
"
`;

exports[`should wrap in module with scoped name 1`] = `
"declare module \\"@company/project\\" {
  declare type Test = \\"ok\\" | \\"error\\";
  declare type Test2 = \\"ok\\" | \\"error\\";
  declare type Maybe<T> =
    | {
        type: \\"just\\",
        value: T,
        ...
      }
    | {
        type: \\"nothing\\",
        ...
      };
  declare type Ref<T> = {
    current: T,
    ...
  };
  declare var ok: number;
}
"
`;
